### PR TITLE
Rails 6 (auto) update

### PR DESCRIPTION
**Pull request by martymcguire**
Most of the stuff here is Rails 6's `rails app:update`, which is mostly bootstrapping, config, and scripts.

There's definitely some pain from the previous Rails 4 => 5 migrate as config things weren't necessarily in the right places.

Also a couple of Rails churn updates. Wrapping some SQL args in `Arel.sql(...)`, deprecating `update_attributes` for `update`, ...

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -14,77 +16,105 @@
...
 
-  # Precompile additional assets.
-  # application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
-  config.assets.precompile += %w( 3d.js )
+  config.action_mailer.perform_caching = false
 
   # Ignore bad email addresses and do not raise email delivery errors.
   # Set this to true and configure the email server for immediate delivery to raise delivery errors.
   # config.action_mailer.raise_delivery_errors = false
-  config.action_mailer.default_url_options = { :host => 'wethebuilders.com' }
```
Comment (lorennorman) [inline config/environments/production.rb:67]: this is gone without replace (at least in this file), could be handled by other configs now though.

File: config/puma.rb [hunk 2]
```diff
@@ -1,15 +1,43 @@
-workers Integer(ENV['WEB_CONCURRENCY'] || 2)
```
Comment (lorennorman) [inline config/puma.rb:1]: important file to double-check for sysadmin stuff

**Review by lorennorman — COMMENTED** [review]
looks good/normal to me, overall. no need to respond to my comments, they're rhetorical to jog your mindset.
good luck! 🚀

File: config/cable.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+development:
+  adapter: async
+
+test:
+  adapter: test
+
+production:
+  adapter: redis
+  url: <%= ENV.fetch("REDIS_URL") { "redis://localhost:6379/1" } %>
+  channel_prefix: epluribus_production
```
[the remark this answers is not in the document]
Comment (martymcguire) [reply config/cable.yml:10]: this was auto-generated by the app upgrade but isn't necessary if we don't use ActionCable (we don't). i'll remove it!

File: config/environments/production.rb [hunk 4]
```diff
@@ -14,77 +16,105 @@
...
+  config.active_storage.service = :local
+
+  # Mount Action Cable outside main process or domain.
+  # config.action_cable.mount_path = nil
+  # config.action_cable.url = 'wss://example.com/cable'
+  # config.action_cable.allowed_request_origins = [ 'http://example.com', /http:\/\/example.*/ ]
 
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
-  config.force_ssl = true
+  # config.force_ssl = true
```
[the remark this answers is not in the document]
Comment (martymcguire) [reply config/environments/production.rb:49]: _great_ catch, thank you!!

File: config/environments/production.rb (hunk 1 above)
Comment (martymcguire) [reply config/environments/production.rb:67]: thanks for spotting this! i had 'fixed' this with another method in another file, but i like _this_ method (in another file). now in `config/initializers/smtp.rb` with the rest of the ActionMailer setup.

File: config/initializers/filter_parameter_logging.rb [hunk 5]
```diff
@@ -1,4 +1,6 @@
 # Be sure to restart your server when you modify this file.
 
 # Configure sensitive parameters which will be filtered from the log file.
-Rails.application.config.filter_parameters += [:password]
+Rails.application.config.filter_parameters += [
+  :passw, :secret, :token, :_key, :crypt, :salt, :certificate, :otp, :ssn
+]
```
[the remark this answers is not in the document]
Comment (martymcguire) [reply config/initializers/filter_parameter_logging.rb:6]: i guess it's now `:passw`? we only allow sign-in through oauth so we don't actually have any form or handler that should receive passwords.

**Comment by martymcguire** [comment]
sooooo many thanks for all this. @lorennorman !

made some changes and double-checked the other things you mentioned. the puma config is a little bit of an open question (it was made for heroku's Procfile-based system and I am not sure how/whether fly.io is making use of that as they use a fly.toml config that doesn't seem to mention either).

I suspect I'll learn quickly what breaks when I launch this tonight or later this week!

**Pull request merged**